rhobs/alerting/data_plane/prometheus.pipeline_to_snapshot_alerts.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: rhtap-pipeline-to-snapshot-exceeded-alerting-rules
+  labels:
+    tenant: rhtap
+spec:
+  groups:
+  - name: pipeline-to-snapshot-exceeded
+    interval: 1m
+    rules:
+    - alert: PipelineRunToSnapshotExceeded
+      expr: |
+       (
+        (
+         sum(increase(integration_svc_response_seconds_bucket{le="+Inf"}[5m])) by (namespace, source_cluster) -
```

Comment:
Is there any difference between the result of `increase(...)` and the result of `sum(increase(...) by (...)`.
I think the current implementation is logically ok, but I don't see the added value with summing the result of `increase`.

Reply:
> Is there any difference between the result of `increase(...)` and the result of `sum(increase(...) by (...)`. I think the current implementation is logically ok, but I don't see the added value with summing the result of `increase`.

I tried that initially and it won't work without 'sum'. I have tried again there now and it just won't calculate.

Reply:
`has_application_failed_deletion_total` in `application_alerts` file has more than one time series per cluster, so there it might be that summing makes sense.

the `by ...` part affects how the time series returned by the query inside the `sum` call will be aggregated. Currently, the label combination we give there is already unique between all the results so it effectively doesn't sum anything.

The reason it won't work for you without `sum` is that the arithmetic operators (`-`, `/`) you use cannot match between the time series that you provide as inputs.

Prometheus will only run operations between time series with identical labels. In our case, we try to subtract and divide between time series that differ between each other by the `le` label, so we need to tell Prometheus either exactly which labels it should try to match or the labels that it should ignore.

Something like this:
```
       (
        (
         increase(integration_svc_response_seconds_bucket{le="+Inf"}[5m])
         - ignoring(le)
         increase(integration_svc_response_seconds_bucket{le="30"}[5m])
        ) / ignoring(le)
        increase(integration_svc_response_seconds_bucket{le="+Inf"}[5m])
       ) > 0.10
```

Reply:
thank you so much for your help and patience.